examples/step-82/step-82.cc

```diff
@@ -0,0 +1,1500 @@
+                                }
+                          }
+
+                        for (unsigned int i = 0; i < n_dofs; ++i)
+                          for (unsigned int j = 0; j < n_dofs; ++j)
+                            {
+                              matrix(local_dof_indices_neighbor[i],
+                                     local_dof_indices_neighbor_2[j]) +=
+                                stiffness_matrix_n1n2(i, j);
+                              matrix(local_dof_indices_neighbor_2[i],
+                                     local_dof_indices_neighbor[j]) +=
+                                stiffness_matrix_n2n1(i, j);
+                            }
+                      } // boundary check face_2
+                  }     // for face_2
+              }         // boundary check face_1
+          }             // for face_1
+
+
+        // Finally, we compute and add the two penalty terms.
+        for (unsigned int face_no = 0; face_no < cell->n_faces(); ++face_no)
+          {
+            const typename DoFHandler<dim>::face_iterator face =
+              cell->face(face_no);
+
+            const double mesh_inv = 1.0 / face->diameter(); // h_e^{-1}
+            const double mesh3_inv =
+              1.0 / std::pow(face->diameter(), 3); // ĥ_e^{-3}
+
+            fe_face.reinit(cell, face_no);
+
+            ip_matrix_cc = 0; // filled in any case (boundary or interior face)
+
+            const bool at_boundary = face->at_boundary();
+            if (at_boundary)
+              {
+                for (unsigned int q = 0; q < n_q_points_face; ++q)
+                  {
+                    const double dx = fe_face.JxW(q);
+
+                    for (unsigned int i = 0; i < n_dofs; ++i)
+                      for (unsigned int j = 0; j < n_dofs; ++j)
+                        {
+                          ip_matrix_cc(i, j) += penalty_jump_grad * mesh_inv *
+                                                fe_face.shape_grad(j, q) *
+                                                fe_face.shape_grad(i, q) * dx;
+                          ip_matrix_cc(i, j) += penalty_jump_val * mesh3_inv *
+                                                fe_face.shape_value(j, q) *
+                                                fe_face.shape_value(i, q) * dx;
+                        }
+                  }
+              }
+            else
+              { // interior face
+
+                const typename DoFHandler<dim>::active_cell_iterator
+                                   neighbor_cell = cell->neighbor(face_no);
+                const unsigned int face_no_neighbor =
+                  cell->neighbor_of_neighbor(face_no);
+
+                if (neighbor_cell->id().operator<(cell->id()))
+                  { // we need to have a global way to compare the cells in
+                    // order to not calculate the same jump term twice
+                    continue; // skip this face (already considered)
```

Comment:
Doxygen cannot handle such comments.
```suggestion
                  { 
                    // we need to have a global way to compare the cells in
                    // order to not calculate the same jump term twice
                    // skip this face (already considered)
                    continue; 
```

Reply:
Thanks for spotting this. What about using the same structure as in the function `compute_errors`, namely

```suggestion
                // In the next step, we need to have a global way to compare the
                // cells in order to not calculate the same jump term twice:
                if (neighbor_cell->id() < cell->id())
                    continue; // skip this face (already considered) 
```

Reply:
Sounds good!